test/CodeGenerator/AbstractInjectorTest.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-di for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-di/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Di\CodeGenerator;
+
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\Container\ContainerInterface;
+use stdClass;
+use Zend\Di\CodeGenerator\AbstractInjector;
+use Zend\Di\CodeGenerator\FactoryInterface;
+use Zend\Di\DefaultContainer;
+use Zend\Di\InjectorInterface;
+use ZendTest\Di\TestAsset\CodeGenerator\StdClassFactory;
+use ZendTest\Di\TestAsset\InvokableInterface;
+
+use function uniqid;
+
+/**
+ * @covers \Zend\Di\CodeGenerator\AbstractInjector
+ */
+class AbstractInjectorTest extends TestCase
+{
+    /**
+     * @var InjectorInterface|ObjectProphecy
+     */
+    private $decoratedInjectorProphecy;
+
+    /**
+     * @var ContainerInterface|ObjectProphecy
+     */
+    private $containerProphecy;
+
+    protected function setUp()
```

Comment:
Please add return types where applicable

Reply:
@Ocramius we can probably take this offline, but do you feel this is necessary in test case classes, particularly with things like `setUp`/`tearDown`, assertions, and test methods, which have established context?

Reply:
Yes: something that declares to be `: void` always has some side-effect, and this makes it clear.

Reply:
Coming from a more haskell-ish world, this is basically our `:: IO ()`